daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala

```diff
@@ -47,6 +47,18 @@ object Ref {
   type ApplicationId = LedgerString
   val ApplicationId: LedgerString.type = LedgerString
 
+  /** Identifiers for participant node users, which act as clients to the Ledger API.
+   *
+   * The concept of participant node users has been introduced after the concept
+   * of client applications, which is why the code contains both of them.
+   * The [[UserId]] identifiers for participant node users are a strict subset of the
```

Comment:
I don’t understand the relation you are sketching out here. I thought your suggestion was that an application id identifies a full Daml App and all clients (which can map to different users) use the same one?

Reply:
The initial idea was that `ApplicationId` becomes equivalent to `UserId` in our codebase. That doesn't really work, as then `UserId` have too wide a character set and in particular allow for spaces, which is a security problem wrt human redability.

So, the proposal here is that on top of the existing custom JWT tokens, which can be used to authenticate requests for any valid `ApplicationId`, the Ledger API server also allows standard JWT tokens whose `sub` claims must be a valid `UserId` to authenticate requests whose `application_id` field matches that `sub` claim.